pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -194,6 +195,12 @@ func (l *LibvirtDomainManager) preStartHook(vm *v1.VirtualMachine, domain *api.D
 		}
 	}
 
+	// setup networking
+	err = network.SetupPodNetwork(vm, domain)
```

Comment:
That still looks like we would retry starting the vm. It is very likely that we can't configure the dhcp server correct anymore, one this failed since the network device is most likely already partly deconfigured. Retrying would be pointless.

Reply:
Ugh... you're right.  It's technically possible this preStartHook to get called twice (for example, if there's an error defining the domain with libvirt after the preStartHook completes it will result in preStartHook occurring again next try)

As it's written now, a second call to SetupPodNetwork will 100% cause virt-launcher to panic.  

Here's how I think we can fix this.

1. Make SetupPodNetwork Idempotent.  This could be done by caching the result at the end of SetupPodNetwork to a flat file and checking to see if that file exists at the beginning of the function. If the file exists, ensure that the domain object has the right info in it and fast succeed.

2. Be certain that any error we return from SetupPodNetwork has the ability to recover in the event of a second execution of SetupPodNetwork.  Any unrecoverable errors should result in virt-launcher exiting.

Reply:
Let's be pragmatic. virt-handler guarantees that it does not try to work on the same vm in parallel. That means we can just touch a file after we have collected the data. We can add something to this file or another file, once we successfully started the dhcpserver the first time. Later invocations can then just skip the network setup. The dhcpserver can panic if something goes wrong.